visualization/app/codeCharta/ui/fileExtensionBar/fileExtensionBar.component.spec.ts

```diff
@@ -120,4 +144,44 @@ describe("FileExtensionBarController", () => {
 			expect(fileExtensionBarController["_viewModel"].isExtensiveMode).toBeTruthy
 		})
 	})
+
+	describe("togglePercentageAbsoluteValues", () => {
+		it("should set viewModel.isAbsoluteValueVisible to false", () => {
+			fileExtensionBarController["_viewModel"].isAbsoluteValueVisible = true
+
+			fileExtensionBarController.togglePercentageAbsoluteValues()
+
+			expect(fileExtensionBarController["_viewModel"].isAbsoluteValueVisible).toBeFalsy()
+		})
+
+		it("should set viewModel.isAbsoluteValueVisible to true", () => {
+			fileExtensionBarController["_viewModel"].isAbsoluteValueVisible = false
+
+			fileExtensionBarController.togglePercentageAbsoluteValues()
+
+			expect(fileExtensionBarController["_viewModel"].isAbsoluteValueVisible).toBeTruthy
+		})
+	})
+
+	describe("highlightBarHoveredBuildings", () => {
+		it("should call findBuildingsSummarizedInOtherDistribution function, when other is given as parameter", () => {
+			fileExtensionBarController["findBuildingsSummarizedInOtherDistribution"] = jest.fn()
```

Comment:
this function remains untested if mocked here. Let's test it here by asserting that we call addBuildingToHighlightList() with the right buildings

Reply:
or x times